tests/unit/configFile.spec.js

```diff
@@ -430,6 +430,49 @@ describe("ConfigFile", () => {
       expect(settings.providersGroups).toEqual(expectedGroups);
     });
 
+    it("Provider groups should be added or removed if override flag is true", async () => {
+      // New group should be added.
+      const newGroups = [
+        ["1", "Group 1"],
+        ["2", "Group 2"],
+        ["3", "Group 3"],
+        ["4", "Group 4"],
+        ["5", "Group 5"],
+      ];
+      await ConfigFile.parseJSONFile(
+        { ...defaultSettings, groups: newGroups },
+        true
+      );
+
+      let settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      let expectedGroups = ConfigFile.parseGroups(newGroups);
+      expect(settings.providersGroups).toEqual(expectedGroups);
+
+      // Groups should be removed if removed in file.
+      await ConfigFile.parseJSONFile(
+        { ...defaultSettings, groups: newGroups.slice(0, 4) },
+        true
+      );
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expectedGroups = ConfigFile.parseGroups(newGroups.slice(0, 4));
+      expect(settings.providersGroups).toEqual(expectedGroups);
+
+      // If new groups are less than 3:
+      // 1) only group name should be updated with new one
+      // 2) disable other groups up to 3
+      // 3) delete all groups greater than 3
+      await ConfigFile.parseJSONFile(
```

Comment:
Just one more change: please separate unit tests for each one of these edge cases (e.g. new groups > current groups; new groups < current groups; new groups < 3; new groups = current groups), and group them all using another describe.

Something like:

```js
describe("parseJSONFile()", () => {
    describe("Groups names should be updated is forced (no matter flag's value)", () => {
         it("Groups names should be updated") // newGroups.length == currentGroups.length

         it("New groups should be added") // newGroups.length > currentGroups.length

         it("Old obsolete groups should be removed") // newGroups.length < currentGroups.length

         it("Undefined groups should be disabled") // newGroups.length < 3

```